lib/AST/Attr.cpp

```diff
@@ -350,9 +350,17 @@ void DeclAttributes::print(ASTPrinter &Printer, const PrintOptions &Options,
   AttributeVector modifiers;
 
   for (auto DA : llvm::reverse(FlattenedAttrs)) {
+    // Always print function builder attribute.
+    bool isFunctionBuilderAttr = false;
+    if (auto *CA = dyn_cast<CustomAttr>(DA)) {
+      if (auto *VD = dyn_cast<ValueDecl>(D)) {
+        isFunctionBuilderAttr = VD->getAttachedFunctionBuilder() == CA;
```

Comment:
`VD->getAttachedFunctionBuilder()` is going to return the same value throughout the loop, I recommend moving it out of the loop to avoid repeating the call for each loop iteration.

Reply:
Done